File: python-client-sdk/src/orbapi/models/shared/event.py
import dataclasses
from dataclasses_json import dataclass_json
from orbapi import utils
from typing import Any, Optional


@dataclass_json
@dataclasses.dataclass
class Event:
    customer_id: str = dataclasses.field(metadata={'dataclasses_json': { 'letter_case': utils.field_name('customer_id') }})
    event_name: str = dataclasses.field(metadata={'dataclasses_json': { 'letter_case': utils.field_name('event_name') }})
    id: str = dataclasses.field(metadata={'dataclasses_json': { 'letter_case': utils.field_name('id') }})
    properties: dict[str, Any] = dataclasses.field(metadata={'dataclasses_json': { 'letter_case': utils.field_name('properties') }})
    timestamp: str = dataclasses.field(metadata={'dataclasses_json': { 'letter_case': utils.field_name('timestamp') }})
    external_customer_id: Optional[str] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('external_customer_id') }})
    

File: go-client-sdk/pkg/models/shared/event.go
package shared

type Event struct {
	CustomerID         string                 `json:"customer_id"`
	EventName          string                 `json:"event_name"`
	ExternalCustomerID *string                `json:"external_customer_id,omitempty"`
	ID                 string                 `json:"id"`
	Properties         map[string]interface{} `json:"properties"`
	Timestamp          string                 `json:"timestamp"`
}
